java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1619,13 +1639,19 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_getNativeColumnView(JNI
 }
 
 JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeEmptyCudfColumn(JNIEnv *env,
-                                                                             jobject j_object,
-                                                                             jint j_type) {
+                                                                             jclass,
+                                                                             jint j_type,
+                                                                             jint scale) {
 
   try {
     cudf::jni::auto_set_device(env);
     cudf::type_id n_type = static_cast<cudf::type_id>(j_type);
-    cudf::data_type n_data_type(n_type);
+    cudf::data_type n_data_type;
+    if ( n_type == cudf::type_id::DECIMAL32 || n_type == cudf::type_id::DECIMAL64 ) {
```

Comment:
This should be a helper function that gets used everywhere.

Reply:
I've an idea that we pass `jobject dtype` instead of type_id and scale. And we create a helper function to convert `jobject dtype` to `cudf::data_type` via JNI. SetIntField. We can also build helper function for transaction from  `cudf::data_type`  to `jobject dtype`  in the same way.

Reply:
> I have an idea that we pass jobject dtype instead of type_id and scale. And we create a helper function who converts jobject dtype to cudf::data_type via JNI.

Please _do not do this_.  Having the JNI call back into the JVM is quite expensive in practice.  It will be much faster to pass the elements of an object to the JNI call rather than an object pointer to JNI and require the JNI function to call back into the JVM to access the instance's fields.

Reply:
@jlowe Thanks for pointing out that. I didn't realize JVM callback in the JNI env is quite inefficient.